modules/pam_time/pam_time.c

```diff
@@ -54,6 +54,18 @@ _pam_parse (const pam_handle_t *pamh, int argc, const char **argv, const char **
     int ctrl = 0;
 
     *conffile = PAM_TIME_CONF;
+
+#ifdef VENDORDIR
+    /* If the specified file does not exist, try the vendor file as fallback. */
+    /* This can be overruled again by a user provided file definition (conffile=...) */
+    struct stat buffer;
+    if (stat(*conffile, &buffer) != 0) {
+      *conffile = VENDORDIR"/security/time.conf";
```

Comment:
The order looks wrong. I think you want to change first conffile before checking that the file exists? Else the error message about a missing file will point to the wrong file.

Reply:
Hm, I think it is correct. The check will be done on PAM_TIME_CONF which should be in this case /etc/security/time.conf. If this does not exists we are switching to /usr/etc/security/time.conf.
The error message should then be:
"cannot read settings from /etc/security/time.conf.. Trying /usr/etc/security/time.conf."
Or does I have overseen something ?

Reply:
Seems I got confused as the next message (LOG_WARNING) sounded like an error message.
Since the fallback is wanted, I don't think we should log a warning, if at all an info message.